lib/Exporter/ONNXModelWriter.cpp

```diff
@@ -748,109 +904,157 @@ llvm::Error ONNXModelWriter::writeBatchBoxCox(const BatchBoxCoxNode *node,
                                                  GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
+  // Add dictionary entries.
   addValueAttribute(proto, "group", node->getGroup());
   addValueAttribute(proto, "kernel", node->getKernel());
 
-  return writeAllWithNode(node->getName(), node, proto);
+  return writeAllWithNode("ChannelShuffle", node, proto);
 }
 
 llvm::Error
 ONNXModelWriter::writeQuantizationProfile(const QuantizationProfileNode *node,
                                           GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
+  // Add dictionary entries.
   addValueAttribute(proto, "name", node->getProfiledNodeName());
   addValueAttribute(proto, "number", node->getProfiledOutputNumber());
 
-  return writeAllWithNode(node->getName(), node, proto);
+  return writeAllWithNode("QuantizationProfile", node, proto);
 }
 
 llvm::Error ONNXModelWriter::writeTraceEvent(const TraceEventNode *node,
                                              GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
+  // Add dictionary entries.
   addValueAttribute(proto, "name", node->getEventName());
   addValueAttribute(proto, "type", node->getEventType());
   addValueAttribute(proto, "index", node->getIndex());
 
-  return writeAllWithNode(node->getName(), node, proto);
+  return writeAllWithNode("TraceEvent", node, proto);
 }
 
 llvm::Error ONNXModelWriter::writeInsertTensor(const InsertTensorNode *node,
                                                GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
+  // Add dictionary entries.
   addValueAttribute(proto, "start", node->getStart());
   addValueAttribute(proto, "count", node->getCount());
   addValueAttribute(proto, "axis", node->getAxis());
 
-  return writeAllWithNode(node->getName(), node, proto);
+  return writeAllWithNode("InsertTensor", node, proto);
 }
 
 llvm::Error ONNXModelWriter::writeChannelwiseQuantizedConvolution(
     const ChannelwiseQuantizedConvolutionNode *node, GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
+  // Add dictionary entries.
   addValueAttribute(proto, "kernel_shape", node->getKernels());
   addValueAttribute(proto, "strides", node->getStrides());
   addValueAttribute(proto, "pads", node->getPads());
   addValueAttribute(proto, "group", node->getGroup());
   addValueAttribute(proto, "group_wise", node->getGroupwise() ? 1 : 0);
 
-  return writeAllWithNode(node->getName(), node, proto);
+  return writeAllWithNode("ChannelwiseQuantizedConvolution", node, proto);
 }
 
 llvm::Error ONNXModelWriter::writeSplat(const SplatNode *node,
                                         GraphType &graph) {
   auto *proto = graph.add_node();
-  // Find dictionary entries.
-  addValueAttribute(proto, "value", node->getValue());
+  // Convert value to tensor with result shape
```

Comment:
So for Splat the proto will include `value` repeated `tensor.size()` times? Why can't we just use `ConstantOfShape` here? Or at least follow the same design as `ConstantOfShape`, where we provide a single `value` attribute and 1D `input` tensor of the shape? Just seems like it's an unnecessary waste of memory.

Reply:
The problem is Max operator, it needs not only the value but the shape as well.